src/app/ReadClient.h

```diff
@@ -240,6 +246,13 @@ class ReadClient : public Messaging::ExchangeDelegate
     ReadClient * GetNextClient() { return mpNext; }
     void SetNextClient(ReadClient * apClient) { mpNext = apClient; }
 
+    // The application has to know to
+    // a) allocate a ReadPrepareParams object that will have fields mpEventPathParamsList and mpAttributePathParamsList with
+    // lifetimes as long as the ReadClient itself and b) free' those up later in the call to OnDeallocatePaths. Note: At a given
+    // time in the system, you can either have a single subscription with re-sub enabled that that has mKeepSubscriptions = false,
+    // OR, multiple subs with re-sub enabled with mKeepSubscriptions = true. You shall not have a mix of both simultaneously.
+    CHIP_ERROR SendAutoResubscribeRequest(ReadPrepareParams && aReadPrepareParams);
```

Comment:
That comment is fine in explaining the constraints, but what does this function do?  Seems like the comment should start with:

```
Like SendSubscribeRequest, but the ReadClient will automatically attempt to re-establish the subscription if
we decide that the subscription has dropped.  The exact behavior of the re-establishment can be controlled
by setting mResubscribePolicy in the ReadPrepareParams.  If not set, a default behavior with exponential
backoff will be used.
```

Reply:
Updated, thanks